src/argilla/client/feedback/schemas.py

```diff
@@ -103,49 +108,80 @@ class QuestionSchema(BaseModel):
     name: str
     title: Optional[str] = None
     description: Optional[str] = None
-    required: Optional[bool] = True
-    settings: Dict[str, Any]
+    required: bool = True
+    settings: Dict[str, Any] = Field(default_factory=dict, allow_mutation=False)
 
     @validator("title", always=True)
     def title_must_have_value(cls, v: Optional[str], values: Dict[str, Any]) -> str:
         if not v:
-            return values["name"].capitalize()
+            return values.get("name").capitalize()
         return v
 
+    class Config:
+        validate_assignment = True
+        extra = Extra.forbid
+
 
 # TODO(alvarobartt): add `TextResponse` and `RatingResponse` classes
 class TextQuestion(QuestionSchema):
-    settings: Dict[str, Any] = Field({"type": "text", "use_markdown": False})
+    settings: Dict[str, Any] = Field({"type": "text", "use_markdown": False}, allow_mutation=False)
     use_markdown: bool = False
 
-    @validator("use_markdown", always=True)
-    def update_settings_with_use_markdown(cls, v: bool, values: Dict[str, Any]) -> bool:
-        if v:
-            values["settings"]["use_markdown"] = v
-        return False
-
-    class Config:
-        validate_assignment = True
-        extra = Extra.forbid
+    @root_validator(skip_on_failure=True)
+    def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        values["settings"]["use_markdown"] = values.get("use_markdown", False)
+        return values
 
 
 class RatingQuestion(QuestionSchema):
-    settings: Dict[str, Any] = Field({"type": "rating"})
-    values: List[int] = Field(unique_items=True)
+    settings: Dict[str, Any] = Field({"type": "rating"}, allow_mutation=False)
+    values: List[int] = Field(unique_items=True, min_items=2)
 
-    @validator("values", always=True)
-    def update_settings_with_values(cls, v: List[int], values: Dict[str, Any]) -> List[int]:
-        if v:
-            values["settings"]["options"] = [{"value": value} for value in v]
-        return v
+    @root_validator(skip_on_failure=True)
+    def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        values["settings"]["options"] = [{"value": value} for value in values.get("values")]
+        return values
 
-    class Config:
-        validate_assignment = True
-        extra = Extra.forbid
+
+class _LabelQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field(default_factory=dict, allow_mutation=False)
+    labels: Union[List[str], Dict[str, str]] = Field(unique_items=True, min_items=2)
+    visible_labels: Optional[conint(ge=3)] = 20
```

Comment:
in the backend we're using `options` instead of `labels`, maybe it's better to align to that here

Reply:
Yes I know, but we decided that user-wise it's better to use `labels` as doesn't make much sense to define a `LabelQuestion` with `options` instead of `labels`, but if you both agree @frascuchon, I can revert that to be `options` too 😄